File: injector/src/test/java/org/codejargon/feather/DependencyTest.java
package org.codejargon.feather;

import org.junit.Test;

import javax.inject.Provider;

import static org.junit.Assert.assertNotNull;

public class DependencyTest {
    @Test
    public void dependencyInstance() {
        Injector injector = Injector.with();
        assertNotNull(injector.instance(Plain.class));
    }

    @Test
    public void provider() {
        Injector injector = Injector.with();
        Provider<Plain> plainProvider = injector.provider(Plain.class);
        assertNotNull(plainProvider.get());
    }

    @Test(expected = InjectorException.class)
    public void unknown() {
        Injector injector = Injector.with();
        injector.instance(Unknown.class);
    }

    public static class Plain {

    }

    public static class Unknown {
        public Unknown(String noSuitableConstructor) {

        }
    }
}

File: photon/src/test/kotlin/net/mready/photon/DependencyTest.kt
package net.mready.photon

import org.junit.Assert.assertNotNull
import org.junit.Test

class DependencyTest {
    @Test
    fun dependencyInstance() {
        val injector = Injector.with()
        assertNotNull(injector.instance(Plain::class.java))
    }

    @Test
    fun provider() {
        val injector = Injector.with()
        val plainProvider = injector.provider(Plain::class.java)
        assertNotNull(plainProvider.get())
    }

    @Test(expected = InjectorException::class)
    fun unknown() {
        val injector = Injector.with()
        injector.instance(Unknown::class.java)
    }

    class Plain

    class Unknown(noSuitableConstructor: String)
}


